test/groovy/com/sap/piper/UtilsTest.groovy

```diff
@@ -43,12 +46,148 @@ class UtilsTest extends BasePiperTest {
         assertThat(result, is('0dad6c33b6246702132454f604dee80740f399ad'))
     }
 
+    @Test
+    void testStashWithDefaults() {
+        Map stashProperties
+
+        def examinee = new Utils()
+        examinee.steps = [
+            stash: { Map stashProps ->
+                stashProperties = stashProps
+            },
+        ]
+        examinee.echo = {}
```

Comment:
The examinee initialization is very similar every time. Wouldn't it be possible to call a function and only pass the required parameters, e.g. only `stashProperties` or the closure?

Reply:
Something like this worked for me:
```
    def newExaminee(Map results) {
        results.stashProperties = null
        def examinee = new Utils()
        examinee.steps = [
            stash: { Map stashProperties ->
                results.stashProperties = stashProperties
            },
        ]
        examinee.echo = {}
        return examinee
    }
```

Reply:
Normally we use `helper.registerAllowedMethod("stash", ...)` to use a instrumented mock for the Jenkins `stash` and `unstash` steps. Since this does not seem to work here, we modify the examinee (`Utils`) and override `stash` and `unstash` with an instrumented mock. Sounds very similar to me.

The examinee implementations look very similar. In most cases we override `stash` or `unstash`, and we silence `echo`.
To cover the differences you could simply pass the closures to the `newExaminee` function (which I proposed).

Reply:
... done, we have now a factory method for our `Utils`-examinee.